__tests__/order/order.mutation.test.js

```diff
@@ -0,0 +1,347 @@
+              ... on Order {
+                _id
+              }
+              ... on Error {
+                statusCode
+                message
+              }
+            }
+          }
+        `,
+        variables: { id: "5f46a8ac90e86913ed0a95d8" },
+      }).catch(err => err);
+
+
+      const error = res;
+      expect(error.graphQLErrors[0].message).toBe("ORDER_NOT_FOUND");
+  });
+
+  test('Should update order', async () => {
+    const order = await client.mutate({
+      mutation: gql`
+        mutation($order: OrderInput!, $id: ID!) {
+          updateOrder(order: $order, id: $id) {
+            ...on Order {
+              _id
+              status
+              totalPrice {
+                currency
+                value
+              }
+              user {
+                address {
+                  appartment
+                  city
+                  region
+                  street
+                  country
+                  buildingNumber
+                  zipcode
+                }
+                email
+                lastName
+                firstName
+                phoneNumber
+              }
+              items {
+                bottomColor {
+                  lang
+                  value
+                }
+                closure {
+                  lang
+                  value
+                }
+                model {
+                  lang
+                  value
+                }
+                closureColor 
+                size{
+                  widthInCm
+                  weightInKg
+                  heightInCm
+                  volumeInLiters
+                  depthInCm
+                }
+                additions {
+                  lang
+                  value
+                }
+                actualPrice {
+                  currency
+                  value
+                }
+                name {
+                  lang
+                  value
+                }
+                pattern {
+                  lang
+                  value
+                }
+                category {
+                  lang
+                  value
+                }
+                quantity
+                colors {
+                  lang
+                  value
+                }
+                subcategory {
+                  lang
+                  value
+                }
+                bottomMaterial {
+                  lang
+                  value
+                }
+              }
+              delivery {
+                sentBy
+                invoiceNumber
+              }
+              paymentMethod
+            }
+            ...on Error {
+              statusCode
+              message
+            }
+          }
+        }
+      `,
+      variables: { order: newOrderUpdated, id: orderId },
+    });
+
+    const updatedOrder = order.data.updateOrder;
+    
+    expect(updatedOrder).toMatchObject(orderUpdateResult);
```

Comment:
It is not the best approach to writing tests. I understand that it is much easier to assert the whole object but it makes tests very unreadable. Please, try to assert at least top-level fields separately. For nested values of fields, it is okay to use objects but for root one - no. Thanks

Reply:
fixed